Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -5,12 +5,34 @@ namespace Chessington.GameEngine.Pieces
 {
     public class Pawn : Piece
     {
-        public Pawn(Player player) 
+        private bool hasMoved = false;
+
+        public Pawn(Player player)
             : base(player) { }
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var location = board.FindPiece(this);
+            var availableLocations = new List<Square>();
+
+            var direction = Player == Player.Black ? 1 : -1;
+
+            // Normal advancing
+            availableLocations.Add(new Square(location.Row + direction*1, location.Col));
```

Comment:
There should be whitespace around the asterisk

Reply:
Yep, fixed in 6c5cbf8